packages/webapp/src/stories/Animals/Details/AnimalDetails.stories.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { Suspense } from 'react';
+import { useForm } from 'react-hook-form';
+import { Meta, StoryObj } from '@storybook/react';
+import AnimalCreationDetails, {
+  AnimalDetailsProps,
+} from '../../../components/Animals/AddAnimalsDetails';
+import AnimalDetails from '../../../components/Animals/AddAnimalsDetails';
+import { FormMethods } from '../../../components/Animals/AddAnimalsDetails/type';
+import {
+  typeOptions,
+  breedOptions,
+  sexOptions,
+  useOptions,
+  tagTypeOptions,
+  tagColorOptions,
+  tagPlacementOptions,
+  organicStatusOptions,
+  originOptions,
+} from './mockData';
+
+// https://storybook.js.org/docs/writing-stories/typescript
+const meta: Meta<AnimalDetailsProps> = {
+  title: 'Components/AddAnimalsDetails',
+  component: AnimalDetails,
+};
+export default meta;
+
+type Story = StoryObj<typeof AnimalCreationDetails>;
+
+export const Default: Story = {
+  render: () => {
+    const formMethods: FormMethods = useForm();
+
+    return (
+      <Suspense>
+        <div style={{ padding: '16px' }}>
+          <AnimalDetails
```

Comment:
If we adopt the pattern in the [MultiStepForm component](https://github.com/LiteFarmOrg/LiteFarm/pull/3196) I think we are going to start wrapping the whole form in `<FormProvider>`.

I _think_ that means we can just use `<FormProvider>` here in the Stories, and always `useFormContext` in the form components rather than passing the `formMethods` prop, and then the components will be that much easier to fit into the way we ultimately do the form (and the fewer props passed will be the bonus!)

But am I missing something? Is there an advantage to passing `formMethods` as prop?

Reply:
It was just my unawareness and misunderstanding, thank you so much!